api/src/anipy_api/provider/base.py

```diff
@@ -131,6 +132,34 @@ def __init_subclass__(cls) -> None:
                     )
                 )
 
+    def generate_new_session(self):
+        """Close the last session, and create a new one.
+
+        Returns:
+            A brand new session
+        """
+        if hasattr(self, "session"):
+            self.session.close()
+        self.session = Session()
+        return self.session
+
+    def request_page(self, req: Request):
+        """Prepare a request and send it, but create a new session if self.session is broken
+        
+        Args:
+            req: The request
+
+        Returns:
+            out: Response of the request
+        """
+        try:
+            return request_page(self.session, req)
+        except RequestConnectionError:
+            # If there is a connection error,
+            # give it a second try with a
+            # new session
+            return request_page(self.generate_new_session(), req)
```

Comment:
All this fix does is create a new session if the old session had a connection error. It only retries once because if that is the problem, one retry will fix it. So, if the issue happens a second time, it'll just raise the error, giving the opportunity for the programmer using the library to catch any actually relevant errors.

Reply:
Ok, all good then.

Reply:
The connection error occurs if the session is not used for a long period of time (for example, watching an episode of anime in between usage ;))